kernel/preemption/Cargo.toml

```diff
@@ -6,7 +6,7 @@ description = "Handles enabling and disabling preemption for each CPU core"
 edition = "2021"
 
 [dependencies]
-cls = { path = "../cls" }
+cls_macros = { path = "../cls/cls_macros" }
```

Comment:
any particular reason why we depend on `cls_macros` directly now? FWIW, i think it's cleaner to depend on `cls` directly and allow the `cls_macros` to be an internal implementation detail.

Reply:
Because depending on `cls` would lead to a circular dependency. That's why the `cls_dep` argument on the macro is necessary.

Reply:
Ah yup, ok.